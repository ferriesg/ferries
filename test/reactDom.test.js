import Enzyme, { mount } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import Admin from "../pages/admin/index";
import jsdom from "jsdom";
import renderer from "react-test-renderer";

// const { JSDOM } = jsdom;
// const { window } = new JSDOM("");
// const { document } = new JSDOM(``).window;


// global.document = document;
// global.window = window;

const { JSDOM } = jsdom;
if (typeof document === 'undefined') {
    const dom=new JSDOM('<!doctype html><html><head></head><body></body></html>');
    global.window =dom.window;
    global.document = global.window.document;
    global.navigator = global.window.navigator;
}

Enzyme.configure({ adapter: new Adapter() });

describe("test new Dom", () => {
  it("go to snapshot", () => {
    const tree = renderer.create(<Admin />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Enzyme shallow dom", () => {
    const warpper = Enzyme.shallow(<Admin />);
    expect(warpper.debug()).toMatchSnapshot();
  });
  it("Enzyme mount dom", () => {
    // sinon.spy(Admin.prototype, 'componentDidMount');
    const mountWarpper = mount(<Admin />);
    console.log(mountWarpper, 2);
  });
  it("Enzyme render dom ", () => {
    const renderWarpper = Enzyme.render(<Admin />);
    expect(renderWarpper).toMatchSnapshot();
  });
});
