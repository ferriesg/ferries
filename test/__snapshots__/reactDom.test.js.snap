// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test new Dom Enzyme render dom  1`] = `
LoadedCheerio {
  "0": <div
    class="adminstyles__AllBox-sc-1yhuc3r-0 iKoYgT"
  >
    <div
      class="adminstyles__SignUpBox-sc-1yhuc3r-1 hxEikv"
    >
      <span
        class="ant-input-affix-wrapper ant-input-affix-wrapper-lg"
        style="margin-bottom:40px"
      >
        <span
          class="ant-input-prefix"
        >
          <span
            aria-label="user"
            class="anticon anticon-user"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="user"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
              />
            </svg>
          </span>
        </span>
        <input
          class="ant-input ant-input-lg"
          placeholder="Email"
          type="text"
          value=""
        />
      </span>
      <span
        class="ant-input-affix-wrapper ant-input-affix-wrapper-lg"
      >
        <span
          class="ant-input-prefix"
        >
          <span
            aria-label="question-circle"
            class="anticon anticon-question-circle"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="question-circle"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
              />
              <path
                d="M623.6 316.7C593.6 290.4 554 276 512 276s-81.6 14.5-111.6 40.7C369.2 344 352 380.7 352 420v7.6c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V420c0-44.1 43.1-80 96-80s96 35.9 96 80c0 31.1-22 59.6-56.1 72.7-21.2 8.1-39.2 22.3-52.1 40.9-13.1 19-19.9 41.8-19.9 64.9V620c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8v-22.7a48.3 48.3 0 0130.9-44.8c59-22.7 97.1-74.7 97.1-132.5.1-39.3-17.1-76-48.3-103.3zM472 732a40 40 0 1080 0 40 40 0 10-80 0z"
              />
            </svg>
          </span>
        </span>
        <input
          class="ant-input ant-input-lg"
          placeholder="PassWord"
          type="password"
          value=""
        />
      </span>
      <div
        style="margin-top:20px;display:flex;justify-content:center;align-items:center"
      >
        <button
          class="ant-btn ant-btn-round ant-btn-primary ant-btn-lg ant-btn-background-ghost"
          style="width:50%"
          type="button"
        >
          <span>
            Log In
          </span>
        </button>
        <span
          class="adminstyles__RegistereBox-sc-1yhuc3r-2 hJHQgI"
        >
          No account? Please register
        </span>
      </div>
    </div>
  </div>,
  "_root": LoadedCheerio {
    "0": Document {
      "children": Array [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`test new Dom Enzyme shallow dom 1`] = `
"<adminstyles__AllBox>
  <adminstyles__SignUpBox>
    <ForwardRef size=\\"large\\" placeholder=\\"Email\\" prefix={{...}} style={{...}} />
    <ForwardRef size=\\"large\\" placeholder=\\"PassWord\\" type=\\"password\\" prefix={{...}} />
    <div style={{...}}>
      <Button type=\\"primary\\" shape=\\"round\\" size=\\"large\\" ghost={true} style={{...}} onClick={[Function: onClick]}>
        Log In
      </Button>
      <adminstyles__RegistereBox onClick={[Function: onClick]}>
        No account? Please register
      </adminstyles__RegistereBox>
    </div>
  </adminstyles__SignUpBox>
</adminstyles__AllBox>"
`;

exports[`test new Dom go to snapshot 1`] = `
<div
  className="adminstyles__AllBox-sc-1yhuc3r-0 iKoYgT"
>
  <div
    className="adminstyles__SignUpBox-sc-1yhuc3r-1 hxEikv"
  >
    <span
      className="ant-input-affix-wrapper ant-input-affix-wrapper-lg"
      onMouseDown={[Function]}
      style={
        Object {
          "marginBottom": "40px",
        }
      }
    >
      <span
        className="ant-input-prefix"
      >
        <span
          aria-label="user"
          className="anticon anticon-user"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="user"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
            />
          </svg>
        </span>
      </span>
      <input
        className="ant-input ant-input-lg"
        hidden={null}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Email"
        style={null}
        type="text"
        value=""
      />
    </span>
    <span
      className="ant-input-affix-wrapper ant-input-affix-wrapper-lg"
      onMouseDown={[Function]}
    >
      <span
        className="ant-input-prefix"
      >
        <span
          aria-label="question-circle"
          className="anticon anticon-question-circle"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="question-circle"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
            />
            <path
              d="M623.6 316.7C593.6 290.4 554 276 512 276s-81.6 14.5-111.6 40.7C369.2 344 352 380.7 352 420v7.6c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V420c0-44.1 43.1-80 96-80s96 35.9 96 80c0 31.1-22 59.6-56.1 72.7-21.2 8.1-39.2 22.3-52.1 40.9-13.1 19-19.9 41.8-19.9 64.9V620c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8v-22.7a48.3 48.3 0 0130.9-44.8c59-22.7 97.1-74.7 97.1-132.5.1-39.3-17.1-76-48.3-103.3zM472 732a40 40 0 1080 0 40 40 0 10-80 0z"
            />
          </svg>
        </span>
      </span>
      <input
        className="ant-input ant-input-lg"
        hidden={null}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="PassWord"
        style={null}
        type="password"
        value=""
      />
    </span>
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "center",
          "marginTop": "20px",
        }
      }
    >
      <button
        className="ant-btn ant-btn-round ant-btn-primary ant-btn-lg ant-btn-background-ghost"
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "width": "50%",
          }
        }
        type="button"
      >
        <span>
          Log In
        </span>
      </button>
      <span
        className="adminstyles__RegistereBox-sc-1yhuc3r-2 hJHQgI"
        onClick={[Function]}
      >
        No account? Please register
      </span>
    </div>
  </div>
</div>
`;
